CMakeLists.txt

```diff
@@ -1,21 +1,46 @@
 # CMakeLists.txt
 cmake_minimum_required(VERSION 3.15)
 project(cpp_template)
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+if(NOT APPLE)
+    # set(CMAKE_CXX_CLANG_TIDY clang-tidy)
```

Comment:
is the `set` needed?

Reply:
This `set` is needed if you want to run `clang-tidy`.
We don't want to run clang tidy on every build since it slows build time a lot.
We should figure out how to do `clang-tidy` properly in another PR. I suspect the best solution is to create another cmake workflow for it so it can be run in CI.